### PR TITLE
use nix-shell to build on nixos

### LINKED ISSUE #616: proposed documentation and shell.nix for building on nixos
# Documentation issue
I needed to add a `shell.nix` with the following contents to successfully build zola on nixos.
```nix
with import <nixpkgs> {};

pkgs.mkShell {
  buildInputs = [
    libsass
    openssl
    pkgconfig
  ];
}
```
## Summary
Needed to add this file, then run `nix-shell` in the cloned directory in order to run `cargo build --release`.

## Proposed solution
I would like to include this `shell.nix`, and update the documentation for nixos users. This will make it easier for anyone who wants to build zola in a nix environment. I can PR for this if you are on board.

### PR DESCRIPTION
**Pull request by ghost**
Fix #616. Add brief documentation for how to use it.

**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [ ] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?

### CODE REVIEW

File: docs/content/documentation/getting-started/installation.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ To build it from source, you will need to have Git, [Rust (at least 1.30) and Ca
 installed. You will also need additional dependencies to compile [libsass](https://github.com/sass/libsass):
 
 - OSX, Linux and other Unix: `make` (`gmake` on BSDs), `g++`, `libssl-dev`
+	- NixOS: Invoke `nix-shell` in the cloned directory. This opens a shell with the above dependencies. You then run the below command to build the release.
```
Comment (Keats) [inline docs/content/documentation/getting-started/installation.md]: looks like it has a space before the `-`?

File: shell.nix [hunk 2]
```diff
@@ -0,0 +1,9 @@
+with import <nixpkgs> {};
+
+pkgs.mkShell {
+  buildInputs = [
+    libsass
+    openssl
+    pkgconfig
+  ];
+}
```
Comment (Keats) [inline shell.nix]: Ah sorry I was thinking more of putting that in the documentation rather than in the repository

**Review by Keats — CHANGES_REQUESTED** [review]

File: docs/content/documentation/getting-started/installation.md (hunk 1 above)
Comment (ghost) [reply docs/content/documentation/getting-started/installation.md]: @Keats indeed, I wanted it to be a subpoint of the above instruction

File: shell.nix (hunk 2 above)
Comment (ghost) [reply shell.nix]: ok, I'll put it as a snippet below the bullet point for nixOS.

**Comment by Keats** [comment]
Thanks!

**Pull request merged**